.github/scripts/validate_scale_config.py

```diff
@@ -0,0 +1,338 @@
+    response.raise_for_status()
+
+    os.makedirs(os.path.dirname(local_filename), exist_ok=True)
+
+    # Write the content to a local file
+    with open(local_filename, "wb") as f:
+        f.write(response.content)
+
+
+def pull_temp_config_from_github_repo(config_path: str) -> str:
+    config_url = GITHUB_PYTORCH_REPO_RAW_URL + config_path
+
+    temp_dir = tempfile.mkdtemp()
+    config_path = os.path.join(temp_dir, config_path)
+    download_file(config_url, config_path)
+
+    return config_path
+
+
+def main() -> None:
+    args = parse_args()
+
+    generate_files = False
+    if args.pytorch_repo_root is None:
+        print(
+            "Using github's pytorch/pytorch repository as the source for the pytorch scale config files"
+        )
+
+        pt_lf_scale_config_path = pull_temp_config_from_github_repo(
+            PYTORCH_LF_SCALE_CONFIG_PATH
+        )
+        pt_lf_canary_scale_config_path = pull_temp_config_from_github_repo(
+            PYTORCH_LF_CANARY_SCALE_CONFIG_PATH
+        )
+    else:
+        # Running locally
+        generate_files = True
+        pt_lf_scale_config_path = os.path.join(
+            args.pytorch_repo_root, PYTORCH_LF_SCALE_CONFIG_PATH
+        )
+        pt_lf_canary_scale_config_path = os.path.join(
+            args.pytorch_repo_root, PYTORCH_LF_CANARY_SCALE_CONFIG_PATH
+        )
+
+    scale_config_path = os.path.join(args.test_infra_repo_root, SCALE_CONFIG_PATH)
+
+    scale_config = load_scale_config(scale_config_path)
+    validation_success = True
+    print(f"Validating internal consistency of {scale_config_path}")
+    if not is_config_consistent_internally(scale_config[RUNNER_TYPE_CONFIG_KEY]):
+        validation_success = False
+        print("scale-config.yml is not internally consistent")
+    print()
+
+    if generate_files:
+        generate_repo_scale_config(
+            scale_config_path, pt_lf_scale_config_path, PREFIX_LF
+        )
+
+        generate_repo_scale_config(
+            scale_config_path, pt_lf_canary_scale_config_path, PREFIX_LF_CANARY
+        )
+        print()
+
+    pt_scale_config = load_scale_config(pt_lf_scale_config_path)
+    pytorch_canary_scale_config = load_scale_config(pt_lf_canary_scale_config_path)
+
+    if not is_consistent_across_configs(
+        scale_config[RUNNER_TYPE_CONFIG_KEY],
+        pt_scale_config[RUNNER_TYPE_CONFIG_KEY],
+        PREFIX_LF,
+    ):
+        print(
+            f"Consistency validation failed between {scale_config_path} and {pt_lf_scale_config_path}"
+        )
+        validation_success = False
+        print()
+
+    if not is_consistent_across_configs(
+        scale_config[RUNNER_TYPE_CONFIG_KEY],
+        pytorch_canary_scale_config[RUNNER_TYPE_CONFIG_KEY],
+        PREFIX_LF_CANARY,
+    ):
+        print(
+            f"Consistency validation failed bewteen {scale_config_path} and {pt_lf_canary_scale_config_path}"
+        )
+        validation_success = False
+        print()
```

Comment:
Why?
```suggestion
```

Reply:
Eh, there's a better way to get the formatting I want...